frontend/pages/SoftwarePage/components/AddSoftwareForm/AddSoftwareForm.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState } from "react";
+
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import Spinner from "components/Spinner";
+import Button from "components/buttons/Button";
+import FileUploader from "components/FileUploader";
+import Graphic from "components/Graphic";
+
+import {
+  getFileDetails,
+  getInstallScript,
+  shouldDisableFormSubmit,
+} from "./helpers";
+import AddSoftwareAdvancedOptions from "../AddSoftwareAdvancedOptions/AddSoftwareAdvancedOptions";
+
+const baseClass = "add-software-form";
+
+const UploadingSoftware = () => {
+  return (
+    <div className={`${baseClass}__uploading-message`}>
+      <Spinner centered={false} />
+      <p>Uploading. It may take few minutes to finish.</p>
+    </div>
+  );
+};
+
+// TODO: if we reuse this one more time, we should consider moving this
+// into FileUploader as a default preview. Currently we have this in
+// AddProfileModal.tsx and here.
+const FileDetails = ({
+  details: { name, platform },
+}: {
+  details: {
+    name: string;
+    platform: string;
+  };
+}) => (
+  <div className={`${baseClass}__selected-file`}>
+    <Graphic name="file-pkg" />
+    <div className={`${baseClass}__selected-file--details`}>
```

Comment:
It feels weird to have `--name`, `--platform` etc using two hyphens as this is not the pattern we've been using in our code since these aren't really modifiers. The reason I asked is we've been pushing for common patterns so I want to make sure we're staying consistent. 

Not a blocker for me, but def a conversation

Reply:
Maybe if we made it it's own component, we can turn `${baseClass}__selected-file` into just `selected-file` and stuff like `${baseClass}__selected-file--details--name` into `selected-file__name`?